pinecone/errors.go

```diff
@@ -0,0 +1,12 @@
+package pinecone
+
+import "fmt"
+
+type PineconeError struct {
```

Comment:
This might be unnecessary, but it's somewhat straightforward to adhere to the `error` interface while also adding additional metadata and information that callers can use if they want. Otherwise, the `Error()` output should represent the entirety of the error.